pygmt/tests/test_rose.py

```diff
@@ -0,0 +1,202 @@
+"""
+Tests for rose.
+"""
+import numpy as np
+import pytest
+from pygmt import Figure
+from pygmt.datasets import load_fractures_compilation
+
+
+@pytest.mark.mpl_image_compare
+def test_rose_data_file():
+    """
+    Test supplying data from sample dataset.
+    """
+
+    data = load_fractures_compilation()
+
+    fig = Figure()
+    fig.rose(
+        data=data,
+        region=[0, 1, 0, 360],
+        sector=15,
+        diameter="5.5c",
+        color="blue",
+        frame=["x0.2g0.2", "y30g30", "+glightgray"],
+        columns=[1, 0],
+        pen="1p",
+        norm="",
+        scale=0.4,
+    )
+    return fig
+
+
+@pytest.mark.mpl_image_compare
+def test_rose_2d_array_single():
+    """
+    Test supplying a 2D numpy array containing a single pair of lengths and
+    directions.
+    """
+
+    data = np.array([[40, 60]])
+
+    fig = Figure()
+    fig.rose(
+        data=data,
+        region=[0, 1, 0, 360],
+        sector=10,
+        diameter="5.5c",
+        color="cyan",
+        frame=["x0.2g0.2", "y30g30", "+glightgray"],
+        pen="1p",
+        norm=True,
+        scale=0.4,
+    )
+
+    return fig
+
+
+@pytest.mark.mpl_image_compare
+def test_rose_2d_array_multiple():
+    """
+    Test supplying a 2D numpy array containing a list of lengths and
+    directions.
+    """
+
+    data = np.array(
+        [[40, 60], [60, 300], [20, 180], [30, 190], [60, 90], [40, 110], [80, 125]]
+    )
+
```

Comment:
We can also define the numpy array as a pytest fixture to reuse it.

Reply:
This numpy array is unique and only defined once so probably don't need a fixture?

Reply:
It's used in two tests.